contracts/protocols/mainnet/uniswap/helpers.sol

```diff
@@ -0,0 +1,351 @@
+        uint24 fee
+    ) internal view returns (address poolAddr) {
+        poolAddr = PoolAddress.computeAddress(
+            nftManager.factory(),
+            PoolAddress.PoolKey({ token0: token0, token1: token1, fee: fee })
+        );
+    }
+
+    function depositAmount(
+        uint256 tokenId,
+        uint256 amountA,
+        uint256 amountB
+    ) internal view returns (uint256 liquidity) {
+        (, , address _token0, address _token1, uint24 _fee, int24 tickLower, int24 tickUpper, , , , , ) = nftManager
+        .positions(tokenId);
+
+        IUniswapV3Pool pool = IUniswapV3Pool(getPoolAddress(_token0, _token1, _fee));
+
+        // compute the liquidity amount
+        {
+            (uint160 sqrtPriceX96, , , , , , ) = pool.slot0();
+            uint160 sqrtRatioAX96 = TickMath.getSqrtRatioAtTick(tickLower);
+            uint160 sqrtRatioBX96 = TickMath.getSqrtRatioAtTick(tickUpper);
+
+            liquidity = LiquidityAmounts.getLiquidityForAmounts(
+                sqrtPriceX96,
+                sqrtRatioAX96,
+                sqrtRatioBX96,
+                amountA,
+                amountB
+            );
+        }
+    }
+
+    function withdrawAmount(uint256 tokenId, uint128 liquidity)
+        internal
+        view
+        returns (uint256 amount0, uint256 amount1)
+    {
+        require(liquidity > 0, "liquidity should greater than 0");
+        (
+            ,
+            ,
+            address _token0,
+            address _token1,
+            uint24 _fee,
+            int24 tickLower,
+            int24 tickUpper,
+            uint128 positionLiquidity,
+            ,
+            ,
+            ,
+
+        ) = nftManager.positions(tokenId);
+
+        require(positionLiquidity >= liquidity, "Liquidity amount is over than position liquidity amount");
+
+        IUniswapV3Pool pool = IUniswapV3Pool(getPoolAddress(_token0, _token1, _fee));
+        (uint160 sqrtPriceX96, int24 currentTick, , , , , ) = pool.slot0();
+        if (currentTick < tickLower) {
+            amount0 = SqrtPriceMath.getAmount0Delta(
+                TickMath.getSqrtRatioAtTick(tickLower),
+                TickMath.getSqrtRatioAtTick(tickUpper),
+                liquidity,
+                false
+            );
+        } else if (currentTick < tickUpper) {
+            amount0 = SqrtPriceMath.getAmount0Delta(
+                sqrtPriceX96,
+                TickMath.getSqrtRatioAtTick(tickUpper),
+                liquidity,
+                false
+            );
+            amount1 = SqrtPriceMath.getAmount1Delta(
+                TickMath.getSqrtRatioAtTick(tickLower),
+                sqrtPriceX96,
+                liquidity,
+                false
+            );
+        } else {
+            amount1 = SqrtPriceMath.getAmount1Delta(
+                TickMath.getSqrtRatioAtTick(tickLower),
+                TickMath.getSqrtRatioAtTick(tickUpper),
+                liquidity,
+                false
+            );
+        }
```

Comment:
@cryptoDev222 I think we can use this function: https://github.com/Uniswap/uniswap-v3-periphery/blob/1f5b385675eee30a77ee10f4dfd6aa0aaa766026/contracts/libraries/LiquidityAmounts.sol#L120